app/src/main/java/com/example/sportsmates/coach/domain/CoachUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.sportsmates.coach.domain
+
+import com.example.sportsmates.UserPreferences
+import com.example.sportsmates.coach.data.Coach
+
+class CoachUseCase(
+    val userPreferences: UserPreferences,
+    private val coachesRepository: CoachesRepository
+) {
+    suspend fun getRelatedCoaches(): List<Coach?> {
+        val sportsList = userPreferences.user?.sportsList ?: emptyList()
+        val coachesList = coachesRepository.getAllCoaches(sportsList)
+        val filteredCoaches = coachesList.filter { coach ->
+            sportsList.contains(coach?.sportName)
+        }.map { coach ->
+            coach?.mainImage = coachesRepository.getCoachMainImage(coach?.coachId)
```

Comment:
use async and await to make a parallel network calls , this will reduce the loading time .
this what i mean -> 
.map { place ->
                coroutineScope {
                    async {
                        place?.mainImage = placesRepository.getPlaceMainImage(place?.placeId)
                        place
                    }
                }
        }
  return filteredPlaces.awaitAll() .